modules/core/src/main/java/org/apache/ignite/internal/util/GridUnsafe.java

```diff
@@ -1587,17 +1595,31 @@ private static Object javaNioAccessObject() {
      * @return Reference to {@code JavaNioAccess.newDirectByteBuffer} method
      * @throws RuntimeException If getting access to the private API is failed.
      */
-    private static Method newDirectBufferMethod(Object nioAccessObj) {
+    private static MethodHandle newDirectBufferMethodHandle(Object nioAccessObj) {
         try {
             Class<?> cls = nioAccessObj.getClass();
 
             Method mtd = cls.getMethod("newDirectByteBuffer", long.class, int.class, Object.class);
 
-            mtd.setAccessible(true);
+            AccessController.doPrivileged((PrivilegedExceptionAction<?>)() -> {
```

Comment:
why is this needed now?

Reply:
It's a good rule to execute all "security-sensitive" operations in privileged sections. Otherwise code may fail when security manager enabled.